test/index-test.js

```diff
@@ -1,30 +1,42 @@
 const expect = require("chai").expect;
-const BaseError = require("../lib/index")["default"];
+const DefaultBaseError = require("../lib/index")["default"];
+const NamedBaseError = require("../lib/index").BaseError;
 
 
 describe("BaseError", () => {
 
-    class TestError extends BaseError {};
+    [
+        ["default export", DefaultBaseError],
+        ["named export", NamedBaseError]
+    ].forEach(([description, BaseError]) => {
 
-    const testError = new TestError("foo");
+        describe(description, () => {
 
-    it("allows subclassing", () => {
-        expect(testError).to.be.an.instanceOf(TestError);
-        expect(testError).to.be.an.instanceOf(BaseError);
-        expect(testError).to.be.an.instanceOf(Error);
-        expect(Error()).to.not.be.an.instanceOf(TestError);
-    });
+            class TestError extends BaseError {}
 
-    it("provides a stack trace", () => {
-        expect(testError.stack).to.be.a("string");
-    });
+            const testError = new TestError("foo");
 
-    it("provides a message", () => {
-        expect(testError.message).to.equal("foo");
-    });
+            it("allows subclassing", () => {
+                expect(testError).to.be.an.instanceOf(TestError);
+                expect(testError).to.be.an.instanceOf(BaseError);
+                expect(testError).to.be.an.instanceOf(Error);
+                expect(Error()).to.not.be.an.instanceOf(TestError);
+            });
+
+            it("provides a stack trace", () => {
+                expect(testError.stack).to.be.a("string");
+            });
+
+            it("provides a message", () => {
+                expect(testError.message).to.equal("foo");
+            });
+
+            it("provides a toString() implementation", () => {
+                expect(testError.toString()).to.equal("Error: foo");
+            });
+
+        });
 
-    it("provides a toString() implementation", () => {
-        expect(testError.toString()).to.equal("Error: foo");
```

Comment:
Yeah, you're right. This is too much. But probably a check for the existence of the default AND named export would make sense. So that the change is represented in the tests somehow? Or do you prefer no tests for that?